src/transformState.js

```diff
@@ -4,8 +4,27 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
 
+      default:
+        for (const key in state) {
+          delete state[key];
+        }
+    }
```

Comment:
extremely risky

If I don't pass anything, I'm more likely to expect an error, or just that nothing will happen, but not that the whole object will be cleared